mobile/src/main/java/ch/epfl/sdp/mobile/state/StatefulSettingsScreen.kt

```diff
@@ -34,7 +56,61 @@ class AuthenticatedUserProfileScreenState(private val user: AuthenticatedUser) :
 fun StatefulSettingsScreen(
     user: AuthenticatedUser,
     modifier: Modifier = Modifier,
+    contentPadding: PaddingValues = PaddingValues(),
 ) {
-  val state = remember(user) { AuthenticatedUserProfileScreenState(user) }
-  SettingsScreen(state, modifier)
+  val chessFacade = LocalChessFacade.current
+  val scope = rememberCoroutineScope()
+  val state = remember(user) { AuthenticatedUserProfileScreenState(user, chessFacade, scope) }
+  SettingsScreen(state, modifier, contentPadding)
+}
+
+/**
+ * Determines the [MatchResult] of a given [Match].
+ *
+ * @param color the [Color] of the current player.
+ * @param match the [Match] to determine its [MatchResult].
+ */
+private suspend fun determineMatchOutcome(
+    color: ch.epfl.sdp.mobile.application.chess.engine.Color,
+    match: Match
+): MatchResult? {
+  val game = match.game.filterNotNull().first()
+  return when (game.nextStep) {
+    is NextStep.Checkmate -> {
+      if (color == (game.nextStep as NextStep.Checkmate).winner) Win(MatchResult.Reason.CHECKMATE)
+      else Loss(MatchResult.Reason.CHECKMATE)
+    }
+    is NextStep.Stalemate -> Tie
+    else -> Tie // Should be changed to null but left for test purposes.
+  }
+}
+
+/**
+ * Returns a [Flow] of a list of [ChessMatch] of the matches played by the given profile.
+ *
+ * @param profile the profile we want to know their played [ChessMatch]es
+ */
+fun chessMatches(profile: Profile, chessFacade: ChessFacade): Flow<List<ChessMatch>> {
+  val matches = chessFacade.matches(profile)
+  return matches.map { list -> list.mapNotNull { match -> match.toChessMatch(profile.uid) } }
```

Comment:
[Question ] May the list of `matches` has null elements and thus requires using `mapNotNull` ?

Reply:
Probably not, you're right. MapNotNull was just used for safety.